src/engine/Actor.ts

```diff
@@ -562,6 +564,25 @@ module ex {
           }
        }
     }
+    /**
+     * Gets the z-index of an actor. The z-index determines the relative order an actor is drawn in.
+     * Actors with a higher z-index are drawn on top of actors with a lower z-index
+     */
+    public getZIndex(): number {
+       return this._zIndex;
+    }
+
+    /**
+     * Sets the z-index of an actor and updates it in the drawing list for the scene. 
+     * The z-index determines the relative order an actor is drawn in.
+     * Actors with a higher z-index are drawn on top of actors with a lower z-index
+     * @param actor The child actor to remove
+     */
+    public setZIndex(newIndex: number) {
+       this.scene._sortedDrawingTree.remove(this);
+       this._zIndex = newIndex;
+       this.scene._sortedDrawingTree.add(this);
```

Comment:
@jedeen I don't think so. You only need to move the lines that modify the tree, you can leave the z-index setting inside `setZIndex` because the order of those operations doesn't appear to affect anything. It could have just as easily been:

```
this._zIndex = newIndex;
this.scene._sortedDrawingTree.remove(this);
this.scene._sortedDrawingTree.add(this);
```

As long as setting the z-index comes before modifying the tree, you're golden.